src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -181,6 +182,8 @@ void QgsLayoutItemLegend::draw( QgsLayoutItemRenderContext &context )
   QgsRenderContext rc = mMap ? QgsLayoutUtils::createRenderContextForMap( mMap, painter, context.renderContext().scaleFactor() * 25.4 )
                         : QgsLayoutUtils::createRenderContextForLayout( mLayout, painter, context.renderContext().scaleFactor() * 25.4 );
 
+  rc.expressionContext().appendScope( QgsExpressionContextUtils::layoutItemScope( this ) );
```

Comment:
Hmm - this is a bit wrong (before the change too) -- we should be also appending all the variables from QgsLayoutItemLegend::createExpressionContext() like "@legend_title", etc.

Reply:
Thanks, done